cmd/telegraf/telegraf.go

```diff
@@ -162,6 +170,39 @@ func reloadLoop(
 	}
 }
 
+// configDirFiles returns a slice made up of filepaths corresponding
+// to config files defined in any of the config directories.
+func configDirFiles() []string {
+	configs := make([]string, 0)
+	for _, dir := range fConfigDirs {
+		walkfn := func(thispath string, info os.FileInfo, _ error) error {
+			if info == nil {
+				log.Printf("W! Telegraf is not permitted to read %s", thispath)
+				return nil
+			}
+
+			if info.IsDir() {
+				return filepath.SkipDir
+			}
+
+			name := info.Name()
+			if len(name) < 6 || name[len(name)-5:] != ".conf" {
+				log.Printf("W! %s is not a valid Telegraf .conf file", name)
+				return nil
+			}
+
+			relPath := filepath.Join(dir, name)
+			configs = append(configs, relPath)
+			return nil
+		}
+
+		if err := filepath.Walk(dir, walkfn); err != nil {
```

Comment:
@srebhan and I reviewed this and had the thought that walking the config dir here might not get the same list of configuration files as the code in config/config.go that initially loaded the files.

Instead of walking the path, could we change Config.LoadConfig to save the list of files? Then we know exactly which files to watch and there's no chance for difference or for a race condition. (Like if a file is added to the dir right after Config.LoadDirectory completes but before this code runs)

Reply:
That's a good point. I don't think I quite have the bandwidth to implement that change - would a Telegraf maintainer be able to take over?

Reply:
@conorevans thank you for working on this, it seems another community member has helped take this over: https://github.com/influxdata/telegraf/pull/12127 if you have time could you check if this change works for you by using the artifacts posted by the bot? thanks!